rdt/transformers/base.py

```diff
@@ -479,3 +479,122 @@ def reverse_transform(self, data):
         data = self._add_columns_to_data(data, reversed_data, self.columns)
 
         return data
+
+
+class BaseMultiColumnTransformer(BaseTransformer):
+    """Base class for all multi column transformers.
+
+    The ``BaseMultiColumnTransformer`` class contains methods that must be implemented
+    in order to create a new multi column transformer.
+
+    Attributes:
+        columns_to_sdtypes (dict):
+            Dictionary mapping each column to its sdtype.
+        prefixes (dict):
+            Dictionary mapping each output column to its prefix.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.columns_to_sdtypes = {}
+        self.prefixes = {}
+
+    def get_input_column(self):
+        """Override ``get_input_column`` method from ``BaseTransformer``.
+
+        Raise an error because for multi column transformers, ``get_input_columns``
+        must be used instead.
+        """
+        raise NotImplementedError(
+            'MultiColumnTransformers does not have a single input column.'
+            'Please use ``get_input_columns`` instead.'
+        )
+
+    def get_input_columns(self):
+        """Return input column name for transformer.
+
+        Returns:
+            list:
+                Input column names.
+        """
+        return self.columns
+
+    def _get_prefix(self):
+        """Return the prefix of the output columns.
+
+        Returns:
+            str:
+                Prefix of the output columns.
+        """
+        raise NotImplementedError()
+
+    def _get_output_to_property(self, property_):
+        self.prefixes = self._get_prefix()
+        is_prefix_dict = isinstance(self.prefixes, dict)
+        output = {}
+        for output_column, properties in self.output_properties.items():
+            # if 'sdtype' is not in the dict, ignore the column
+            if property_ not in properties:
+                continue
+
+            if is_prefix_dict:
+                prefix = self.prefixes[output_column]
+            else:
+                prefix = self.prefixes
```

Comment:
Why is this allowing multiple types for the prefixes? I think it could be complicated if developers have to set custom prefixes and output names in 2 places. Maybe it would be simpler to say that if you want to have custom prefixes for each output name, then just use custom output names and set the prefix to None.

In other words, `_get_prefix` always returns None or a string. And then you can just do the lines below